src/vm/jitinterface.cpp

```diff
@@ -2637,32 +2624,39 @@ void CEEInfo::embedGenericHandle(
         if (pResolvedToken->tokenType == CORINFO_TOKENKIND_Ldtoken)
             pMD = MethodDesc::FindOrCreateAssociatedMethodDescForReflection(pMD, th, methodInst);
 
-        pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)pMD;
+        if (pResult)
+        {
+            pResult->handleType = CORINFO_HANDLETYPE_METHOD;
+            pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)pMD;
+        }
         pTemplateMD = pMD;
 
         // Runtime lookup is only required for stubs. Regular entrypoints are always the same shared MethodDescs.
         fRuntimeLookup = pMD->IsWrapperStub() &&
             (pMD->GetMethodTable()->IsSharedByGenericInstantiations() || TypeHandle::IsCanonicalSubtypeInstantiation(methodInst));
     }
-    else
-    if (!fEmbedParent && pResolvedToken->hField != NULL)
+    else if (!fEmbedParent && pResolvedToken->hField != NULL)
     {
         FieldDesc * pFD = (FieldDesc *)pResolvedToken->hField;
         TypeHandle th(pResolvedToken->hClass);
 
-        pResult->handleType = CORINFO_HANDLETYPE_FIELD;
-
-        pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)pFD;
+        if (pResult)
+        {
+            pResult->handleType = CORINFO_HANDLETYPE_FIELD;
+            pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)pFD;
+        }
 
         fRuntimeLookup = th.IsSharedByGenericInstantiations() && pFD->IsStatic();
     }
     else
     {
         TypeHandle th(pResolvedToken->hClass);
 
-        pResult->handleType = CORINFO_HANDLETYPE_CLASS;
-
-        pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)th.AsPtr();
+        if (pResult)
+        {
+            pResult->handleType = CORINFO_HANDLETYPE_CLASS;
+            pResult->compileTimeHandle = (CORINFO_GENERIC_HANDLE)th.AsPtr();
+        }
 
         if (fEmbedParent && pResolvedToken->hMethod != NULL)
```

Comment:
I doubt that this case is version resilient ... maybe better to fail on it for now.